test/test_overrides.py

```diff
@@ -44,12 +58,435 @@ def gemm_diag(mat1, mat2, out=None):
 
     return DiagonalTensor(mat1._N, mat1._i * mat2._i)
 
-
-
 t1 = DiagonalTensor(5, 1)
 t2 = DiagonalTensor(5, 2)
+t3 = DiagonalTensor(5, 1)
 
 print(gemm_diag(t1, t2))
 print(torch.gemm(t1, t2))
 print(torch.gemm(t1.tensor(), t2.tensor()))
 print(torch.mm(t1.tensor(), t2.tensor()))
+
+class TestOverride(unittest.TestCase):
+    def test_gemm_equality(self):
+        self.assertEqual(t1, t3)
+
+    def test_gemm_diag(self):
+        self.assertEqual(torch.gemm(t1, t2), t2)
+
+
+requires_torch_function = pytest.mark.skipif(
+    not TORCH_FUNCTION_ENABLED,
```

Comment:
we don't want to include this envvar, so you can delete this and the use of `requires_torch_function` everywhere